src/pokeradar.c

```diff
@@ -0,0 +1,590 @@
+#include "unk_0201CED8.h"
+#include "unk_0202D7A8.h"
+#include "unk_02039C80.h"
+#include "unk_0203E880.h"
+#include "unk_020508D4.h"
+#include "unk_02054D00.h"
+#include "unk_020553DC.h"
+#include "unk_0205DAC8.h"
+#include "unk_0205E7D0.h"
+#include "unk_02061804.h"
+#include "pokeradar.h"
+#include "unk_020711EC.h"
+#include "unk_0207D3B8.h"
+#include "overlay005/ov5_021F2D20.h"
+#include "overlay006/ov6_022430C4.h"
+
+#define NUM_GRASS_PATCHES 4
+#define RADAR_BATTERY_STEPS 50
+
+typedef struct {
+    int unk_00;
+    int unk_04;
+    int shakeType;
+    BOOL active;
+    BOOL continueChain;
+    BOOL shiny;
+    UnkStruct_ov101_021D5D90 * unk_18;
+    VecFx32 position;
+} GrassPatch;
+
+typedef struct RadarChain_t {
+    int shakeType;
+    int count;
+    int species;
+    int level;
+    BOOL active;
+    BOOL unk_14;
+    BOOL unk_18;
+    GrassPatch patch[NUM_GRASS_PATCHES];
+    UnkStruct_0201CFEC unk_BC;
+    u8 unk_D0;
+} RadarChain;
+
+static BOOL CheckTileIsGrass(UnkStruct_0203CDB0 * param0, const fx32 param1, const int param2, const int param3, const u8 param4, const u8 param5, GrassPatch * patch);
+static BOOL sub_020698AC(const RadarChain * chain, const int param1, const int param2, u8 * param3);
+static void sub_020698E4(UnkStruct_0203CDB0 * param0, RadarChain * chain);
+static u8 sub_0206994C(UnkStruct_0203CDB0 * param0);
+static BOOL CheckPatchContinueChain(const u8 patchRing, const int battleResult);
+static BOOL CheckPatchShiny(const int param0);
+static void IncWithCap(int * param0);
+
+RadarChain * CreateRadarChain (const int param0) {
+    RadarChain * chain;
+
+    chain = Heap_AllocFromHeap(param0, sizeof(RadarChain));
+
+    sub_0201CFEC(FX32_ONE * 16, FX32_ONE * 8, FX32_ONE * 16, &chain->unk_BC);
+    return chain;
+}
+
+void FreeRadarChain (RadarChain * chain) {
+    Heap_FreeToHeap(chain);
+}
+
+void ClearRadarChain (RadarChain * chain) {
+    u8 patchRing;
+
+    chain->count = 0;
+    chain->shakeType = 0;
+    chain->species = 0;
+    chain->level = 0;
+    chain->active = FALSE;
+    chain->unk_D0 = 0;
+    chain->unk_14 = 1;
+    chain->unk_18 = 0;
+
+    MI_CpuClear8(&(chain->patch[0]), sizeof(GrassPatch) * 4);
+
+    for (patchRing = 0; patchRing < NUM_GRASS_PATCHES; patchRing++) {
+        chain->patch[patchRing].active = FALSE;
+    }
+}
+
+BOOL RadarSpawnPatches (UnkStruct_0203CDB0 * param0, const int param1, const int param2, RadarChain * chain) {
+    u8 patchRing;
+    u8 v1, v2;
+    u8 v3;
+    u8 v4;
+    int v5, v6;
+    u8 v7;
+    const VecFx32 * v8;
+    u8 ringTileCount[4] = {  // Number of tiles in each ring of the radar. Lowest being the most outer ring
+        32, 24, 16, 8
+    };
+
+    v8 = sub_0205EAFC(param0->unk_3C);
+
+    v7 = 0;
+
+    for (patchRing = 0; patchRing < NUM_GRASS_PATCHES; patchRing++) {
+        BOOL v10;
+
+        v3 = inline_020564D0(ringTileCount[patchRing]);
+
+        v1 = 9 - (patchRing * 2);
+
+        v2 = 9 - (patchRing * 2);
+
+        v4 = v3 / v1;
```

Comment:
In general, feel free to condense lines like these to reduce overall whitespace.